lib/gooddata/lcm/actions/rename_existing_client_projects.rb

```diff
@@ -0,0 +1,79 @@
+# encoding: UTF-8
+#
+# Copyright (c) 2010-2017 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require_relative 'base_action'
+
+module GoodData
+  module LCM2
+    class RenameExistingClientProjects < BaseAction
+      DESCRIPTION = 'Rename Existing Client Projects'
+
+      PARAMS = define_params(self) do
+        description 'Client Used for Connecting to GD'
+        param :gdc_gd_client, instance_of(Type::GdClientType), required: true
+
+        description 'Clients'
+        param :clients, array_of(instance_of(Type::HashType)), required: true, generated: true
+
+        description 'Organization Name'
+        param :organization, instance_of(Type::StringType), required: true
+      end
+
+      RESULT_HEADER = [
+        :id,
+        :pid,
+        :old_title,
+        :new_title
+      ]
+
+      class << self
+        def call(params)
+          client = params.gdc_gd_client
+
+          domain_name = params.organization || params.domain
+          domain = client.domain(domain_name) || fail("Invalid domain name specified - #{domain_name}")
+
+          results = []
+          params.clients.each do |c|
+            begin
+              domain_client = domain.clients(c[:id])
+            rescue
+              next
+            end
+
+            begin
+              c_project = domain_client.project
```

Comment:
Please correct me if I am mistaken but if there's 1000 clients, then `domain_client.project` is called 1000x. That means 1K API requests which will degrade performance and overuse the API. Or am I missing something?